.github/workflows/build.yml

```diff
@@ -0,0 +1,302 @@
+            dist/latest.yml
+  # most of that could be done on a Linux worker to speed things up, but
+  # - universal binary building is currently only supported on macOS
+  # - dmg building via electron builder only works on macOS; theoretically
+  #   a dmg can be manually built on Linux, but it's a hacky mess
+  build_sign_m:
+    name: Build MacOS
+    runs-on: macOS-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+      - name: Store package version in environment
+        run: |
+          PACKAGE_VERSION=`jq -r '.version' package.json`
+          echo "PACKAGE_VERSION=$PACKAGE_VERSION" >> $GITHUB_ENV
+      # the signing endpoint requires whitelisting IPs - we use wireguard to
+      # make sure we exit with a known IP. To make things easier the secret
+      # should contain the full client config, including the client secret
+      # key
+      - name: Configure wireguard
+        run: |
+          brew install wireguard-go wireguard-tools
+          echo "${{ secrets.WG_CONFIG }}" | base64 -d | sudo tee /usr/local/etc/wireguard/wg0.conf >/dev/null
+          sudo wg-quick up wg0
+          sudo wg
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - name: Install dependencies
+        run: npm ci --legacy-peer-deps
+      - name: Build the bundle
+        run: npm run build
+      # Producing universal binaries currently fails due to the native realm.node
+      # binary, and embedding multi-arch binaries there currently seems to be
+      # mostly unsupported. Once that has been solved the next few build steps
+      # can be switched to universal. Doing universal builds changes the name
+      # of the installers - but still creates an autoupdate file with the same
+      # name, but new installers referenced - so it should be possible to use
+      # that to pull existing installations to the universal binary via the
+      # update mechanism
+      - name: Prepare installer contents
+        run: |
+          touch node_modules/realm/binding.gyp
+          npx electron-builder -m --publish never --dir
+          #npx electron-builder -m --publish never --dir --universal
+      - name: Prepare signer upload
+        run: |
+          cd dist/mac && zip --symlinks -r "../../Realm Studio.app.zip" "Realm Studio.app"
+          #cd dist/mac-universal && zip --symlinks -r "../../Realm Studio.app.zip" "Realm Studio.app"
+      - name: Download signer
+        run: |
+          wget https://macos-notary-1628249594.s3.amazonaws.com/releases/client/v3.8.1/darwin_amd64.zip
+          unzip darwin_amd64.zip
+      - name: Run signer
+        run: |
+          ./darwin_amd64/macnotary -f "Realm Studio.app.zip" -m notarizeAndSign -u https://dev.macos-notary.build.10gen.cc/api -s "${{ secrets.NOTARY_SECRET }}" -k "${{ secrets.NOTARY_KEY_ID }}" -b com.mongodb.realm-studio -o "output.zip" -t app -e resources/entitlements.mac.plist
```

Comment:
Ditto my two other comments.

Reply:
I'm asking the relevant guys  to be sure - my impression so far was that it doesn't have to be secret. At least the signer URL is already referenced by other mongodb repositories.